src/mlpack/core/tree/binary_space_tree/ub_tree_split.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file ub_tree_split.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Definition of UBTreeSplit, a class that splits the space according
+ * to the median address of points contained in the node.
+ */
+#ifndef MLPACK_CORE_TREE_BINARY_SPACE_TREE_UB_TREE_SPLIT_HPP
+#define MLPACK_CORE_TREE_BINARY_SPACE_TREE_UB_TREE_SPLIT_HPP
+
+#include <mlpack/core.hpp>
+#include "../address.hpp"
+
+namespace mlpack {
+namespace tree /** Trees and tree-building procedures. */ {
+
+template<typename BoundType, typename MatType = arma::mat>
+class UBTreeSplit
+{
+ public:
+  //! The type of a one-dimensional address.
+  typedef typename std::conditional<sizeof(typename MatType::elem_type) * CHAR_BIT <= 32,
+                                    uint32_t,
+                                    uint64_t>::type AddressElemType;
+
+  /**
+   * This class performs the actual splitting i.e. it reorders the dataset.
+   * This variable should be equal to true if the class does not perform the
+   * actual splitting.
+   */
+  static constexpr bool NeedRearrangeDataset = false;
```

Comment:
Oh.. I thought that could solve the problem with the overload, I've forgotten to remove that.